### PR TITLE
DATACOUCH-138 - Support Page/Slice/PagingAndSorting

### PR DESCRIPTION
**Pull request by simonbasle**

### CODE REVIEW

File: src/main/java/org/springframework/data/couchbase/repository/query/support/N1qlUtils.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+  /**
+   * A converter that can be used to extract the {@link CouchbasePersistentProperty#getFieldName() fieldName},
+   * eg. when one wants a path from {@link PersistentPropertyPath#toDotPath(Converter)} made of escaped field names.
+   */
+  public static final Converter<? super CouchbasePersistentProperty,String> FIELD_NAME_ESCAPED =
+      new Converter<CouchbasePersistentProperty, String>() {
+        @Override
+        public String convert(CouchbasePersistentProperty source) {
+          return "`" + source.getFieldName() + "`";
```
Comment (daschl) [inline src/main/java/org/springframework/data/couchbase/repository/query/support/N1qlUtils.java]: you could use our i() helper here

Comment (simonbasle) [inline src/main/java/org/springframework/data/couchbase/repository/query/support/N1qlUtils.java]: figured I'd skip the extraneous Expression creation by not using `i()`.

**Comment by simonbasle** [comment]
cleaned up the commits (fixing SimpleCouchbaseRepository having findAll(...) after local rebase + added javadoc). pinging @daschl for review before squashing and merging :boom:

**Comment by daschl** [comment]
@simonbasle :up:

**Pull request merged**